src/pybind/mgr/dashboard/controllers/settings.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import cherrypy
+
+from . import ApiController, RESTController, AuthRequired
+from ..settings import Settings, Options
+
+
+@ApiController('/settings')
+@AuthRequired()
+class SettingsController(RESTController):
+    @staticmethod
+    def _to_native(setting):
+        return setting.upper().replace('-', '_')
+
+    def list(self):
+        settings = [(name, value) for name, value in Options.__dict__.items()
+                    if name.isupper()]
+        return [{
+            'name': name,
+            'default': default,
+            'type': data_type.__name__,
+            'value': getattr(Settings, name)
+        } for name, (default, data_type) in settings]
+
+    def get(self, name):
+        attr = self._to_native(name)
+        try:
+            default, data_type = getattr(Options, attr)
+        except AttributeError:
+            raise cherrypy.HTTPError(404)
```

Comment:
Consider to use ``DashboardException(http_status_code=404)`` here.

Reply:
I'd use `cherrypy.NotFound()`